src/components/FormSelectionContainer/FormSelect/FormSelect.tsx

```diff
@@ -14,38 +15,80 @@ interface FormSelectProps {
   const { config } = useConfigContext();
   const [errorMsg, setErrorMsg] = useState<string | null>(null);
-  const [isValidDns, setIsValidDns] = useState<boolean>();
+  const [isValidDns, setIsValidDns] = useState<boolean>(true);
+  const [isValidOwner, setIsValidOwner] = useState<boolean>(true);
+  const [isValidEntry, setIsValidEntry] = useState<boolean>(true);
   const refButton = useRef<HTMLDivElement>(null);
 
   const handleForm = async (): Promise<void> => {
+    if (isValidEntry) {
+      onAddForm();
+    } else {
+      ReactTooltip.show(refButton.current as unknown as Element);
+    }
+  };
+
+  const dnsCheck = useCallback(async () => {
     if (config?.network === "local") {
       setIsValidDns(true); // for local e2e to pass, skip dns validate + set valid
     } else {
       const isDnsValidated = await validateDns(form);
-      if (!isDnsValidated) {
-        setErrorMsg(
-          `${getIssuerAddress(form.defaults)} address could not be found at ${getIssuerLocation(
-            form.defaults
-          )} DNS TXT records.`
-        );
-        ReactTooltip.show(refButton.current as unknown as Element);
-      }
       setIsValidDns(isDnsValidated);
     }
-  };
+  }, [config, form]);
 
-  useEffect(() => {
-    if (isValidDns) {
-      onAddForm();
+  const ownershipCheck = useCallback(async () => {
+    const wallet = config?.wallet;
+    const contractAddress = getIssuerAddress(form.defaults);
+    const isDID = checkDID(form.defaults);
+    if (isDID) {
+      setIsValidOwner(true);
+      return;
     }
-  }, [isValidDns, onAddForm]);
+    const contractType = form?.type;
+    if (config?.network === "local") {
+      setIsValidOwner(true); // for local e2e to pass, skip ownership validate + set valid
+    } else if (contractAddress != undefined && wallet != undefined) {
+      const valid = await checkOwnership(contractType, contractAddress, wallet);
+      setIsValidOwner(valid);
+    }
+  }, [config, form]);
+
+  const checkValidity = useCallback(async () => {
+    setIsValidEntry(isValidDns && isValidOwner);
+    const errorMessage: string[] = [];
+    if (!isValidDns) {
+      const contractAddress = getIssuerAddress(form.defaults);
+      errorMessage.push(
+        `The contract address ${contractAddress} could not be found on ${getIssuerLocation(
+          form.defaults
+        )} DNS TXT records.`
+      );
+    }
+    if (!isValidOwner) {
+      const contractAddress = getIssuerAddress(form.defaults);
+      const address = await config?.wallet?.getAddress();
+      errorMessage.push(`The contract ${contractAddress} does not belong to ${address}.`);
+    }
+    if (errorMessage.length > 0) {
+      setErrorMsg(errorMessage.join("<br />"));
+    } else {
+      setErrorMsg(null);
+    }
+  }, [isValidDns, isValidOwner, form, config]);
+
+  useEffect(() => {
+    dnsCheck();
+    ownershipCheck();
+    checkValidity();
+  }, [dnsCheck, ownershipCheck, checkValidity]);
```

Comment:
these will immediately check all forms with dnsprove (google dns), previously was intentional to only do checks on user initiated click on form

Reply:
wdyt, should we stick with onclick or useeffect ?
if useEffect, 
it's average case is 2 call (dnsprove, etherscan) while waiting on load (assuming it's a transferable record not using DID)
worst case is 4 call (dnsprove, 3x etherscan)

Reply:
number of calls multiplied by number of forms also, think stick with onclick? compromise ux for functional abit, unless there are actual user feedback

so maybe need to show some [loader](https://ui.tradetrust.io/?path=/story/ui-loader--spinner) meanwhile